### PR TITLE
Fix for PHP8; no count() on stdClass

### PR DESCRIPTION
**Pull request by nwp90**
PHP8 does not allow count() on stdClass, so initialise haschildren to null, check for that and use a more complicated test for emptiness on haschildren.

### CODE REVIEW

**Comment by mwehr** [comment]
Voting for integration

File: questions.php [hunk 1]
```diff
@@ -187,7 +187,7 @@
                 // just remove the dependency and inform the user about it.
                 $haschildren = $questionnaire->get_all_dependants($qid);
             }
-            if (count($haschildren) != 0) {
+            if ($haschildren !== null and (count($haschildren->directs) + count($haschildren->indirects)) > 0) {
```
Comment (mchurchward) [inline questions.php:190]: The term "and" should be replaced by "&&".

File: questions.php [hunk 2]
```diff
@@ -69,7 +69,7 @@
 }
 
 $questionnairehasdependencies = $questionnaire->has_dependencies();
-$haschildren = [];
+$haschildren = null;
```
Comment (mchurchward) [inline questions.php:72]: This line can be deleted completely, in combination with the suggestions below.

File: questions.php (hunk 1 above)
Comment (mchurchward) [inline questions.php:190]: The block starting at "if ($questionnairehasdependencies) {" can be simplified down to:

```
            $action = "confirmdelquestion";
            if ($questionnairehasdependencies) {
                // Important: due to possibly multiple parents per question
                // just remove the dependency and inform the user about it.
                $haschildren = $questionnaire->get_all_dependants($qid);
                if (count($haschildren->directs) + count($haschildren->indirects) > 0) {
                    $action = "confirmdelquestionparent";
                }
            }

```
The $haschildren will always be a stdClass with "directs" and "indirects" properties, so we can just check the "questionnairehasdependencies" boolean to determine the rest.

**Review by mchurchward — CHANGES_REQUESTED** [review]
Can you apply the code changes I have highlighted?
I would recommend creating a new branch from MOODLE_311_STABLE and resubmitting a new PR in order to get a clean set of commits.

File: questions.php (hunk 1 above)
Comment (mchurchward) [reply questions.php:190]: This was already implemented in MOODLE_400_STABLE as part of https://github.com/PoetOS/moodle-mod_questionnaire/pull/442. I will backport it.

**Pull request closed without merging**